### PR TITLE
Use iterators and sequences

### PR DESCRIPTION
**Pull request by blink1073**
Avoid giving back raw arrays as return values.

### CODE REVIEW

File: src/kernel/manager.ts [hunk 1]
```diff
@@ -81,13 +93,14 @@ class KernelManager implements Kernel.IManager {
...
+    return Kernel.listRunning(this._getOptions(options)).then(it => {
+      let running = toArray(it);
       if (!deepEqual(running, this._running)) {
-        this._running = running.slice();
-        this.runningChanged.emit(running);
+        this._running = running;
+        this.runningChanged.emit(new Vector(this._running));
       }
-      return running;
+      return iter(running);
```
Comment (afshin) [inline src/kernel/manager.ts]: Can return `it` instead of `iter(running)` or is there a mutability issue there?

File: src/session/default.ts [hunk 2]
```diff
@@ -501,7 +505,7 @@ namespace Private {
       }
     }
     return listRunning(options).then(models => {
-      for (let model of models) {
+      for (let model of toArray(models)) {
```
Comment (afshin) [inline src/session/default.ts]: Maybe `each` instead of `for`.

File: src/session/manager.ts [hunk 3]
```diff
@@ -84,13 +96,14 @@ class SessionManager implements Session.IManager {
...
+    return Session.listRunning(this._getOptions(options)).then(it => {
+      let running = toArray(it);
       if (!deepEqual(running, this._running)) {
-        this._running = running.slice();
-        this.runningChanged.emit(running);
+        this._running = running;
+        this.runningChanged.emit(new Vector(running));
       }
-      return running;
+      return iter(running);
```
Comment (afshin) [inline src/session/manager.ts]: Same comment about returning `it` vs. `iter(running)`.

File: src/kernel/manager.ts (hunk 1 above)
Comment (blink1073) [inline src/kernel/manager.ts]: We've already exhausted the iterator in `toArray`

**Comment by afshin** [comment]
👍

**Pull request merged**